### DISCUSSION TITLE
How to use eGPU with AMD 7900xtx in Windows to get the GPU acceleration for AL/ML?

### DISCUSSION THREAD

**Discussion by Yuz2023** [ROCm/ROCm#3003] [Q&A] [answered]
Greetings! I am a newbie here and wondering how I can use eGPU with AMD 7900xtx in Windows to get the GPU acceleration for AL/ML? I switched from rtx 3080 to 7900xtx, the gaming experience is great (PC recognizes it and runs with it well, eGPU is Razer Core X, also has enough power budget) but now I like to try some coding.
I followed the online info. like (https://phazertech.com/tutorials/rocm.html), but encounter with:
•	amdgpu-pro 20.45 amdgpu-dkms fails/error
•	ROCk module is NOT loaded, possibly no GPU devices
I have been searching everywhere and find little talk about my situation (I know my setting may be quite strange and eGPU is not ideal for leveraging the full power of the card, but eGPU is currently what I prefer to go for.
 
Thanks

**Comment by idkwhodatis** [comment]
try directml or zluda for now, ai libs for rocm on windows is still working in progress

**Reply by Yuz2023** [reply]
Thanks for the tips, I am trying in WSL2 but not so sure if it would work or not. Maybe eventually I will need to switch to Linux.

**Reply by idkwhodatis** [reply]
wsl2 prob won't work, there's no GPU virtualization

**Comment by darren-amd** [comment, marked as answer]
Hi @Yuz2023,

The instructions for installing ROCm on Linux vs WSL are a bit different, please try this guide: [Install ROCm for WSL](https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/wsl/install-radeon.html) and let me know if you run into any issues. For ML workloads, you can find a guide to install Pytorch on WSL here: [PyTorch on WSL](https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/wsl/install-pytorch.html).
